elastic/kibana
# [Upgrade Assistant] Migrate server to new es-js client

Part of https://github.com/elastic/kibana/issues/73973

This PR migrates Upgrade Assistant to use the new ES JS client.

### How to test:
These changes are pretty widespread. I think UA has good test coverage to catch regressions, but it would also be good to run through the UI and do a quick smoke test.

1. Hack `x-pack/plugins/upgrade_assistant/server/lib/es_migration_apis.ts` and seed data:

```
      indices: indices.concat([
      {
        level: 'none',
        message: 'none level test',
        url: 'test/url',
        details: 'test details',
        index: 'test1',
        reindex: true,
        needsDefaultFields: false,
        blockerForReindexing: 'index-closed',
      },
      {
        level: 'info',
        message: 'info level test',
        url: 'test/url',
        details: 'test details',
        index: 'test2',
        reindex: true,
        needsDefaultFields: false,
      },
      {
        level: 'warning',
        message: 'warning level test2',
        url: 'test/url',
        details: 'test details',
        index: 'test3',
        reindex: true,
        needsDefaultFields: false,
      },
      {
        level: 'critical',
        message: 'metricbeat-1 successful fix',
        url: 'test/url',
        details: 'test details',
        index: 'metricbeat-1',
        reindex: false,
        needsDefaultFields: true,
      },
      {
        level: 'critical',
        message: 'metricbeat-2 failed fix',
        url: 'test/url',
        details: 'test details',
        index: 'metricbeat-2',
        reindex: false,
        needsDefaultFields: true,
      },
    ]),
```

Then, via Console, create indices:
```
PUT test1/_doc/1
{"foo":"bar"}
PUT test2/_doc/1
{"foo":"bar"}
PUT test3/_doc/1
{"foo":"bar"}
PUT metricbeat-1/_doc/1
{"foo":"bar"}
```

2. Navigate to Upgrade Assistant in Stack Management.
3. Verify enable/disable deprecation logging on the Overview tab
4. Go to the Indices tab and run reindex against `test2` index
